substrate/frame/ranked-collective/src/benchmarking.rs

```diff
@@ -56,131 +58,304 @@ fn make_member<T: Config<I>, I: 'static>(rank: Rank) -> T::AccountId {
+
+		// Create the `Call` for demoting the member.
 		let call = Call::<T, I>::demote_member { who: who_lookup };
-	}: { call.dispatch_bypass_filter(origin)? }
-	verify {
+
+		// Dispatch the call and measure its execution.
+		#[block]
+		{
+			call.dispatch_bypass_filter(origin)?;
+		}
+
+		// Ensure the member's rank has decreased by 1.
 		assert_eq!(Members::<T, I>::get(&who).map(|x| x.rank), rank.checked_sub(1));
+
+		// Ensure the member count remains as expected.
 		assert_eq!(MemberCount::<T, I>::get(rank), 2);
+
+		// Ensure the index of the last member has changed.
 		assert_ne!(last_index, IdToIndex::<T, I>::get(rank, &last).unwrap());
-		assert_last_event::<T, I>(match rank {
-			0 => Event::MemberRemoved { who, rank: 0 },
-			r => Event::RankChanged { who, rank: r - 1 },
-		}.into());
+
+		// Ensure the correct event was emitted depending on the member's rank.
+		assert_last_event::<T, I>(
+			match rank {
+				0 => Event::MemberRemoved { who: who.clone(), rank: 0 },
+				r => Event::RankChanged { who: who.clone(), rank: r - 1 },
+			}
+			.into(),
+		);
+
+		Ok(())
 	}
 
-	vote {
-		let class = T::Polls::classes().into_iter().next().unwrap();
-		let rank = T::MinRankOfClass::convert(class.clone());
+	#[benchmark]
+	fn vote() -> Result<(), BenchmarkError> {
+		// Find the first class or use a default one if none exists.
+		let (class_exists, class) = T::Polls::classes()
+			.into_iter()
+			.next()
+			.map(|c| (true, c))
+			.unwrap_or_else(|| (false, Default::default()));
 
+		// Convert the class to a rank and create a caller based on that rank.
+		let rank = T::MinRankOfClass::convert(class.clone());
 		let caller = make_member::<T, I>(rank);
-		let caller_lookup = T::Lookup::unlookup(caller.clone());
 
-		let poll = T::Polls::create_ongoing(class).expect("Must always be able to create a poll for rank 0");
+		// If the class exists, create an ongoing poll, otherwise use a default poll.
+		let poll = if class_exists {
+			T::Polls::create_ongoing(class.clone())
+				.expect("Should always be able to create a poll for rank 0")
+		} else {
+			Default::default()
+		};
+
+		// Benchmark the vote logic.
+		#[block]
+		{
+			let vote_result =
+				Pallet::<T, I>::vote(SystemOrigin::Signed(caller.clone()).into(), poll, true);
+
+			// If the class exists, expect a successful vote, otherwise expect a `NotPolling` error.
+			if class_exists {
+				assert_ok!(vote_result);
+			} else {
+				assert_err!(vote_result, crate::Error::<T, I>::NotPolling);
+			}
+		}
+
+		// Vote again with a different decision (false).
+		let vote_false =
+			Pallet::<T, I>::vote(SystemOrigin::Signed(caller.clone()).into(), poll, false);
+
+		if class_exists {
+			assert_ok!(vote_false);
+		} else {
+			assert_err!(vote_false, crate::Error::<T, I>::NotPolling);
+		}
+
+		// Vote again with a different decision (false).
+		assert_ok!(Pallet::<T, I>::vote(SystemOrigin::Signed(caller.clone()).into(), poll, false));
```

Comment:
Copy paste mistake?

Reply:
oversight, it's the logic before testing with the secretary collective.